src/CMakeLists.txt

```diff
@@ -260,6 +261,11 @@ if (NOT cereal_FOUND)
   install(TARGETS cereal EXPORT openMVG-targets)
   set(OpenMVG_USE_INTERNAL_CEREAL ON)
 endif()
+if (TARGET cereal::cereal)
+  set(cereal_TARGET "cereal::cereal")
```

Comment:
I've quickly tested if `OpenMVG_Samples/cameras_undisto_Brown` works when linking with external `OpenMVG`
```
project(test)
cmake_minimum_required(VERSION 2.80)

find_package(OpenMVG REQUIRED)

add_executable(openMVG_sample_cameras_undistoBrown undistoBrown.cpp)
target_link_libraries(openMVG_sample_cameras_undistoBrown
  OpenMVG::openMVG_geometry
  OpenMVG::openMVG_image
  OpenMVG::openMVG_multiview
  OpenMVG::openMVG_stlplus
set_property(TARGET openMVG_sample_cameras_undistoBrown PROPERTY FOLDER OpenMVG/Samples)
```
 It complains about missing `cereal::cereal` but also about missing `OpenMP::OpenMP_CXX` target.
```
-- ----
-- OpenMVG Find_Package
-- ----
-- Found OpenMVG version: 2.0.0
-- Installed in: /usr
-- ----
-- Configuring done
CMake Warning (dev) at /usr/share/openMVG/cmake/OpenMVGTargets.cmake:56 (set_target_properties):
  Policy CMP0028 is not set: Double colon in target name means ALIAS or
  IMPORTED target.  Run "cmake --help-policy CMP0028" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  The link interface of target "OpenMVG::cereal" contains:

    cereal::cereal

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /usr/share/openMVG/cmake/OpenMVGConfig.cmake:86 (include)
  CMakeLists.txt:4 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /usr/share/openMVG/cmake/OpenMVGTargets.cmake:110 (set_target_properties):
  Policy CMP0028 is not set: Double colon in target name means ALIAS or
  IMPORTED target.  Run "cmake --help-policy CMP0028" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  The link interface of target "OpenMVG::openMVG_image" contains:

    OpenMP::OpenMP_CXX

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /usr/share/openMVG/cmake/OpenMVGConfig.cmake:86 (include)
  CMakeLists.txt:4 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /home/bartus/.cache/aurutils/sync/openmvg-git/src/openMVG/src/openMVG_Samples/cameras_undisto_Brown/build
```
With `find_package({OpenMP,cereal} REQUIRED)` it works fine
My CMake scripting foo isn't strong at all but I will try to incorporate relevant changes.

Reply:
Looks ok here, not quite sure about other internal/external deps: `flann,cpl,osi,lemon,etc.`

Reply:
I've massaged the samples code a bit to make it work with external `openMVG` and it looks like everything works flowlesly, besides the missing `software\SfM\SfMPlyHelper.hpp` header in one example.
https://github.com/bartoszek/openMVG/commit/c9c7b0a13e807e00f2453dca96c5f2f809d17667

Reply:
Only COMPONENT (library imported with `::`) needs to be added in the cmake.in, else other would work since they have a strong type (static or dynamic). COMPONENT does not encode directly the type of lib, that's why you need to reuse the find_package -> or find_dependency as suggested by cmake

Reply:
Regarding the `software\SfM\SfMPlyHelper.hpp` I agree we should move this file to openMVG/geometry or openMVG/sfm/utils and so we should be able to leverage this file for installed openMVG configuration. Thank you for reporting this! This file SfMPlyHelper.hpp I think is mature enough now to be transitioned to the library.

Appreciated your feedback, and questions to help us make OpenMVG easier and robust to build by anyone